opencog/scm/utilities.scm

```diff
@@ -1019,8 +1023,63 @@
 	; but I think it helps. Do it twice; once is sometimes not enough.
 	(gc) (gc))
 
+
+; ---------------------------------------------------------------------
+(define-public (check-name? node-name node-type)
+"
+ Return #t if there is a node of type node-type with a name "node-name".
+"
+        (not (null? (cog-node node-type node-name)))
+)
+
+; ---------------------------------------------------------------------
+
+(define-public (random-string str-length)
+"
+ random-string -- Returns a random string of length 'str-length'.
+"
+	(define alphanumeric "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789")
+	(define str "")
+	(while (> str-length 0)
+		(set! str (string-append str (string (string-ref alphanumeric (random (string-length alphanumeric))))))
+		(set! str-length (- str-length 1))
+	)
+	str
+)
+
 ; ---------------------------------------------------------------------
 
+(define-public (random-node-name node-type random-length prepend-text)
+"
+ Creates a possible name 'node-name' of length 'random-length' for a node
+ of type 'node-type'. The 'node-name' is not used with any other node
+ of type 'node-type'. Prepend 'prepend-text' to the front.
+"
+	(define node-name (random-string random-length))
+	(define prepend-length (string-length prepend-text))
+	(if (> prepend-length 0)
+		(set! node-name (string-append prepend-text node-name))
+	)
+	(while (check-name? node-name node-type)
+		(if (> prepend-length 0)
+			(set! node-name (string-append prepend-text (random-string random-length)))
+			(set! node-name (random-string random-length))
+		)
+	)
+	node-name
+)
+
+; -----------------------------------------------------------------------
+
+(define-public (choose-var-name)
+"
+ Creates name for VariableNodes after checking whether the name is being
+ not used by other VariableNode.
+"
+    (random-node-name 'VariableNode 36 "$")
+)
```

Comment:
https://github.com/opencog/atomspace/tree/master/opencog/rule-engine#things-need-to-be-implemented and https://groups.google.com/forum/#!topic/opencog/dKCYL47fpCQ.